src/SdfGenerator.cc

```diff
@@ -422,6 +429,41 @@ namespace sdf_generator
     return true;
   }
 
+  void updateModelElementWithNestedInclude(sdf::ElementPtr &_elem,
+                                           const bool _saveFuelVersion)
+  {
+    sdf::ElementPtr e = _elem->GetFirstElement(), nextE = nullptr;
+    while (e != nullptr)
+    {
+      nextE = e->GetNextElement();
+
+      if (e->GetIncludeElement() != nullptr)
+      {
+        if (_saveFuelVersion && e->FilePath().find("fuel") != std::string::npos)
```

Comment:
Wouldn't it be better to check if the file path is in the `includeUriMap`? Searching for the string `fuel` seems a bit brittle.

Reply:
I agree 57d583390c6c9d3f39f520f0570a9637f5060fcd